src/main/java/commands/SearchCommand.java

```diff
@@ -93,29 +203,80 @@ private void printSearchHeading() throws InvalidCommandException {
         }
     }
 
+    /**
+     * Prints the heading of search result base on the category of userAction.
+     *
+     * @param category The category that will be searched.
+     */
     private void printHeadingMessage(String category) {
-        System.out.println("The " + category + "(s)" + " containing keywords"
-                + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
-                + "is(are) listed below.");
-        ui.printLine();
+        switch (userAction) {
+        case SEARCH_WORKOUT_ACTION_KEYWORD:
+            try {
+                Integer.parseInt(userArguments);
+                System.out.println("The " + category + "(s) with "
+                        + ui.getColorText(TextColor.COLOR_YELLOW, "reps = " + userArguments)
+                        + " is(are) listed below.");
+                ui.printLine();
+            } catch (NumberFormatException e) {
+                System.out.println("The " + category + "(s)" + " containing keywords"
+                        + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
+                        + "is(are) listed below.");
+                ui.printLine();
+            }
+            break;
+        default:
+            System.out.println("The " + category + "(s)" + " containing keywords"
+                    + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
+                    + "is(are) listed below.");
+            ui.printLine();
+        }
     }
```

Comment:
Would it be feasible to extract this to the `UI.java` class? I was thinking since this is a UI-related method, would be better to give it to the UI class to handle.

Reply:
However, for this issue, I am afraid that it may be hard to move this method to UI since moving it to UI may need to pass several parameters to the UI class

Reply:
(For future reference) The topic was taken to Telegram and resolved there.